Kubernetes 1.31.1 added to kind tests

rhr323 opened the pull request:
...wip...

rhr323 commented:
buildkite test this -f p=kind,DEPLOYER_CLIENT_VERSION=0.24.0,DEPLOYER_KIND_NODE_IMAGE=kindest/node:v1.31.1@sha256:cd224d8da58d50907d1dd41d476587643dad2ffd9f6a4d96caf530fb3b9a5956 -m DEPLOYER_KIND_IP_FAMILY=ipv4,DEPLOYER_KIND_IP_FAMILY=ipv6

mauri870 commented:
@rhr323 Would you mind also updating the [docs](https://github.com/search?q=repo%3Aelastic%2Fcloud-on-k8s+1.30&type=code) as well? Thanks

thbkrkr commented on README.md:
```diff
@@ -17,7 +17,7 @@ Current features:
 Supported versions:
 
-*  Kubernetes 1.26-1.30
+*  Kubernetes 1.26-1.31
```
```suggestion
*  Kubernetes 1.27-1.31
```

thbkrkr commented on docs/supported-versions.asciidoc:
```diff
@@ -1,4 +1,4 @@
-* Kubernetes 1.26-1.30
+* Kubernetes 1.26-1.31
```
```suggestion
* Kubernetes 1.27-1.31
```

thbkrkr commented on hack/operatorhub/templates/csv.tpl:
```diff
@@ -284,7 +284,7 @@ spec:
 
 
-    * Kubernetes 1.26-1.30
+    * Kubernetes 1.26-1.31
```
```suggestion
    * Kubernetes 1.27-1.31
```

thbkrkr reviewed (approved):
LGTM, just a doubt for the minimum k8s version supported. It seems to me that we have adopted the policy of starting from version N-4.

rhr323 commented:
> LGTM, just a doubt for the minimum k8s version supported. It seems to me that we have adopted the policy of starting from version N-4.

Ok! I updated the docs to start from 1.27. I also removed 1.26 from the nightly test matrix (I hope it's also ok).
